### ISSUE TITLE
blurry buttons

### ISSUE THREAD

**Issue by schiessle** [nextcloud/server#12294]
Don't know it is just me, but the buttons look really blurry on my system (both on the internal monitor of the laptop and on the external), some examples:

* Password enforce dialog on public links:
![image](https://user-images.githubusercontent.com/1589737/48029829-913a8e80-e14f-11e8-84d3-81f70daf7bd6.png)

* Admin security setting "enforce two factor"
![image](https://user-images.githubusercontent.com/1589737/48029875-ac0d0300-e14f-11e8-89c1-87abaa3551a3.png)

Basically all rounded blue buttons are quite blurry on the edges, the gray buttons look good:

![image](https://user-images.githubusercontent.com/1589737/48029960-e2e31900-e14f-11e8-8981-b35798321aa4.png)

**Comment by schiessle** [comment]
cc @nextcloud/designers

**Comment by skjnldsv** [comment]
You mean the antialiasing?

**Comment by MariusBluem** [comment]
Yes.

**Comment by skjnldsv** [comment]
Unfortunately we cannot really do much about this.
The only thing would be to make sure the background goes under the border with `background-clip: border-box;`

The difference is tiny but noticeable (same background, only change is the `background-clip` property.

| Before | After |
|:---------:|:------:|
|![dev skjnldsv com_settings_admin_security 1](https://user-images.githubusercontent.com/14975046/48073483-26d12f00-e1df-11e8-9e87-699ed5d6732c.png)|![dev skjnldsv com_settings_admin_security 2](https://user-images.githubusercontent.com/14975046/48073484-2769c580-e1df-11e8-8cc9-4c5691673a1d.png)|

**Comment by MariusBluem** [comment]
@schiessle and me just tested this with Firefox and even IE11 in a Windows-VM. Same design issue/behaviour. Maybe we should go back to the old buttons - better stable, known and looking good than new, fancy and full of issues 🤔

**Comment by skjnldsv** [comment]
> Maybe we should go back to the old buttons - better stable, known and looking good than new, fancy and full of issues

No preferences here :)
@jancborchardt ?

**Comment by MorrisJobke** [comment]
> @schiessle and me just tested this with Firefox and even IE11 in a Windows-VM. Same design issue/behaviour. Maybe we should go back to the old buttons - better stable, known and looking good than new, fancy and full of issues 🤔

👍 I would say that this is a blocker for the beta 1 - we should not change this after beta 1.

**Comment by juliushaertl** [comment]
How about removing the white border from the primary buttons? This seems to at least improve the antialiasing a bit.

Before (Firefox):
![image](https://user-images.githubusercontent.com/3404133/48083030-6ce4bd80-e1f4-11e8-9683-ebce1fd30275.png)

After (Firefox):

![image](https://user-images.githubusercontent.com/3404133/48083014-63f3ec00-e1f4-11e8-86f4-6242d0ba35e2.png)

**Comment by MariusBluem** [comment]
> How about removing the white border from the primary buttons? This seems to at least improve the antialiasing a bit.

Awesome! I think this would fix this in a nice way ... What do you think @schiessle 🤔

**Comment by jancborchardt** [comment]
I think removing the white border as @juliushaertl said is the best call.

We only need these white borders in 2 cases right now: On the log in screen, and when a button is in the header. In other words, when the background is not the default (white).

For that we could add an additional class like `.button-border`?

**Comment by skjnldsv** [comment]
> For that we could add an additional class like `.button-border`?

I'll rather not. We need to keep our design simple, instead of adding too many classes ;)

**Comment by jancborchardt** [comment]
Ok, then smth like this (quick note, did not test!):
```css
/* Apply border to primary button if on log in page (and not in a dark container) or if in header */
#body-login :not(body-login-container) .button-primary,
#header .button-primary  {
  border: 1px solid #fff;
}
```

**Comment by jancborchardt** [comment]
And we remove the border from buttons completely otherwise, also from non-primary ones.

What do you think?

**Comment by skjnldsv** [comment]
Though if the primary color is too luminous, we'll need the border again :(

**Comment by jancborchardt** [comment]
Or a very slight, very light box-shadow?

**Comment by skjnldsv** [comment]
We could trick the border with an inset shadow yes. Could work :)
But I doubt shadow on the buttons would look good.

**Comment by skjnldsv** [comment]
Any new here?
last call to make a decision is this week!

**Comment by MorrisJobke** [comment]
@jancborchardt What to do here?

**Comment by jancborchardt** [comment]
I will work on a pull request to remove the border like mentioned at https://github.com/nextcloud/server/issues/12294#issuecomment-436585033

**Comment by MorrisJobke** [comment]
> I will work on a pull request to remove the border like mentioned at [#12294 (comment)](https://github.com/nextcloud/server/issues/12294#issuecomment-436585033)

But then the download button in the header of the public page does not look like a button anymore (it's blue on blue background). 🤔

**Comment by jancborchardt** [comment]
> But then the download button in the header of the public page does not look like a button anymore (it's blue on blue background).

It does, as the rule I wrote specifically only sets the border to white for primary buttons on the log in screen or in the header.

Pull request at https://github.com/nextcloud/server/pull/12558, please review (and help with the theming question …)
